### PR TITLE
Adds Gradle 8.0 support

### PR DESCRIPTION
**Pull request by TheHypnoo**
Hi!

As discussed in issue #32 when a project with Gradle 8 does not work the compilation with this PR is solved.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
-## 1.1.2
+
+# CHANGELOG
+
```
Comment (TrangLeQuynh) [inline CHANGELOG.md]: Please delete these extra lines

File: android/build.gradle [hunk 2]
```diff
@@ -25,6 +25,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
+    namespace 'com.tranglequynh.flutter_upgrade_version'
```
Comment (TrangLeQuynh) [inline android/build.gradle]: This didn't work with AGP < 4.2. To ensure you need one more condition:
```
/// Conditional for compatibility with AGP < 4.2
if (project.android.hasProperty("namespace")) {
    namespace 'com.tranglequynh.flutter_upgrade_version'
}
```

**Comment by TheHypnoo** [comment]
@TrangLeQuynh Both comments resolved.

**Comment by TrangLeQuynh** [comment]
Your pull request have 4 commits. Please merge into 1 commit.
Screenshot 2023-12-12 at 17 20 34

**Pull request merged**